serde/src/de/value.rs

```diff
@@ -359,6 +373,7 @@ where
     }
 }
 
+// just why does this not use the `primitive_deserializer` macro??????
```

Comment:
Because it has special handling for `deserialize_enum`, which other primitived does not.
```suggestion
```